extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -99,12 +100,13 @@ export class AmpAd3PImpl extends AMP.BaseElement {
   }
 
   renderOutsideViewport() {
-    const allowRender = allowRenderOutsideViewport(this.element, this.win);
-    if (allowRender !== true) {
-      return allowRender;
+    if (is3pThrottled(this.win)) {
+      return false;
     }
     // Otherwise the ad is good to go.
-    return super.renderOutsideViewport();
+    const elementCheck = getAmpAdRenderOutsideViewport(this.element);
+    return typeof elementCheck == 'number' ?
+      elementCheck : super.renderOutsideViewport();
```

Comment:
Hmm... right now getAmpAdRenderOutsideViewport will never return false (either true or 1.25) so this is not currently an issue but I do agree that how this is setup is somewhat fragile.  I reworked so that getAmpAdRenderOutsideViewport now returns ?number where null indicates use default.  Let me know what you think.